server/controllers/availability.js

```diff
@@ -0,0 +1,99 @@
+const Availability = require("../models/availability");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+// @route POST /schedule/
+// @desc Create a schedule
+// @access Public
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+  const profileId = req.profile.id;
+  const { name, days } = req.body;
+  const schedule = await Availability.findOne({ name, petSitterId: req.profile.id });
+  if (schedule) {
+    res.status(400);
+    throw new Error("Schedule name already exist");
+  }
+  const newSchedule = new Availability.create({
+    profileId,
+    name,
+    days,
+  });
+  if (newSchedule) {
+    if (!req.profile.activeSchedule) {
+      req.profile.set({ activeSchedule: newSchedule.id });
+      await req.profile.save();
+    }
+    res.status(201).json({
+      success: {
+        newSchedule
+      }
+    });
+  } else {
+    res.status(500);
+    throw new Error("Invalid data");
+  }
+});
+
+// @route GET /availability/:scheduleId
+// @desc Get schedule
+// @access Private
+exports.getScheduleId = asyncHandler(async (req, res, next) => {
+  const scheduleId = req.params.scheduleId;
+  const schedule = await Availability.findOne({ _id: scheduleId })
```

Comment:
Let's also make sure that only the logged in user can access their availability schedule

Reply:
done the changes.

Reply:
Added a clause to check if the req.user.id is present in the profile. If not throw an error "not authorized". else allow the user to access their schedule.